src/asciidoc/docs/customizations/media.adoc

```diff
@@ -0,0 +1,220 @@
+
+The following code sample shows all the available `typeOptions`.
+Each option is described in the sections that follow the sample.
+
+[source,json]
+----
+{
+    "name": "Media Upload",
+    "type": "media",
+    "typeOptions": {
+        "acceptOnly": ["image", "video", "audio"],
+        "returnType": "url"|"extended",
+        "hideExtensions": false,
+        "videoOptions": {
+            "hls": true,
+            "estimatePoses": false
+        },
+        "imageOptions": {
+            "width": 100,
+        },
+        "audioOptions": {},
+        "fileOptions": {}
+    }
+}
+----
+
+=== acceptOnly
+
+`acceptOnly` allows you to specify the VCC types that are allowed.
+The value for `acceptOnly` is an array of one or more VCC types.
+If the array contains more than one type, then the user can choose any file of a type that matches one of the types in the array.
+For example, `"acceptOnly": ["image", "video"]` allows the user to select an image file or a video file.
+If the value of `acceptOnly` is an empty array, or if `acceptOnly` is not included in `typeOptions`, then any file type will be allowed.
+
+The possible types are as follows.
+
+* *image* - If you set `acceptOnly` to an array containing `image`, the user will be prompted to choose an image file for the VCC.
+Using a `media` VCC with `acceptOnly` set to `["image"]` is equivalent to using an <<image#, image VCC>>.
+
+* *video* - If you set `acceptOnly` to an array containing `video`, the user will be prompted to choose a video file for the VCC.
+Using a `media` VCC with `acceptOnly` set to `["video"]` is equivalent to using a <<video#, video VCC>>.
+
+* *audio* - If you set `acceptOnly` to an array containing `audio`, the user will be prompted to choose a sound file for the VCC.
+Using a `media` VCC with `acceptOnly` set to `["audio"]` is equivalent to using a <<sound#, sound VCC>>.
+
+* *file* - If you set `acceptOnly` to an array containing `file`, the user will be prompted to choose a file for  the VCC.
+The file selected can be of any type.
+Using a `media` VCC with `acceptOnly` set to `["file"]` is equivalent to using a <<file#, file VCC>>.
+
+=== hideExtensions
+
+The `media` VCC type enables you to hide all asset packs and VCC extensions in cases where they don't make sense (for example, templates for selling premium assets).
+To use this feature in your templates, enable the `hideExtensions` type option.
+In the following code sample, extensions are hidden, so *Browse Image Asset Packs* will not appear as a selection.
+
+[source,json]
+----
+{
+    "key": "hideExtensions",
+    "name": "Image or Video No Extensions",
+    "type": "media",
+    "typeOptions": {
+        "acceptOnly": ["image", "video"],
+        "hideExtensions": true,
+        "videoOptions": {
+            "hls": true
+        }
+    }
+}
+----
+
+The above sample code will display the following selections.
+
+image::mediaNoExtensionsVCC.png[title="Media No Extensions VCC"]
+
+=== videoOptions
```

Comment:
Several things going on here...

* Why is videoOptions the only typeOption for the underlying types that has its own section? We need to add descriptions for the audioOptions, imageOptions, and fileOptions.
* Repeating all of the type options for the underlying types on this page will be hard to maintain as options are added going forward. It might be better to provide examples of the common options and then refer to the underlying type reference for a comprehensive list of options. This pattern might also help clarify the relationship between media and other VCC types.
* Again, simplify this text and make it clear what the videoOptions option is before jumping into the details. Something like, "Sets the configuration options for video values."